src/aks-preview/azext_aks_preview/_client_factory.py

```diff
@@ -0,0 +1,65 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+from azure.cli.core.commands.client_factory import get_mgmt_service_client
+from azure.cli.core.profiles import ResourceType
+from azure.cli.core.profiles import CustomResourceType
+
+CUSTOM_MGMT_AKS_PREVIEW = CustomResourceType('azext_aks_preview.vendored_sdks.azure_mgmt_preview_aks',
+                                             'ContainerServiceClient')
+CUSTOM_MGMT_AKS = CustomResourceType('azext_aks_preview.vendored_sdks.azure_mgmt_aks',
+                                     'ContainerServiceClient')
+
+
+def cf_compute_service(cli_ctx, *_):
+    return get_mgmt_service_client(cli_ctx, ResourceType.MGMT_COMPUTE)
+
+
+def cf_container_services(cli_ctx, *_):
+    return get_container_service_client(cli_ctx).container_services
+
+
+def get_container_service_client(cli_ctx, **_):
+    return get_mgmt_service_client(cli_ctx, CUSTOM_MGMT_AKS)
+
+
+def cf_managed_clusters(cli_ctx, *_):
+    return get_mgmt_service_client(cli_ctx, CUSTOM_MGMT_AKS_PREVIEW)
+
+
+def cf_resource_groups(cli_ctx, subscription_id=None):
+    return get_mgmt_service_client(cli_ctx, ResourceType.MGMT_RESOURCE_RESOURCES,
+                                   subscription_id=subscription_id).resource_groups
+
+
+def cf_resources(cli_ctx, subscription_id=None):
+    return get_mgmt_service_client(cli_ctx, ResourceType.MGMT_RESOURCE_RESOURCES,
+                                   subscription_id=subscription_id).resources
+
+
+def get_auth_management_client(cli_ctx, scope=None, **_):
+    import re
+
+    subscription_id = None
+    if scope:
+        matched = re.match('/subscriptions/(?P<subscription>[^/]*)/', scope)
```

Comment:
re.match matches only from the beginning of the string, is this what is expected? if scope for example doesn't start on /sub... i believe it will fail.